src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeStartUpTests.cs

```diff
@@ -0,0 +1,215 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.EslintBridgeClient
+{
+    [TestClass]
+    public class EslintBridgeStartUpTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<EslintBridgeStartUp, IEslintBridgeStartUp>(null, new[]
+            {
+                MefTestHelpers.CreateExport<string>("some path", EslintBridgeStartUp.EslintBridgeDirectoryMefContractName),
+                MefTestHelpers.CreateExport<INodeLocator>(Mock.Of<INodeLocator>()),
+                MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>())
+            });
+        }
+
+        [TestMethod]
+        public void Start_FailsToFindNode_FileNotFoundException()
+        {
+            var nodeLocator = SetupNodeLocator(null);
+
+            var testSubject = CreateTestSubject(nodeLocator: nodeLocator.Object);
+            Func<Task> act = async () => await testSubject.Start();
+
+            act.Should().ThrowExactly<FileNotFoundException>().And.Message.Should().Contain("node.exe");
+        }
+
+        [TestMethod]
+        public void Start_FailsToFindStartupScript_FileNotFoundException()
+        {
+            const string filePath = "somefile.txt";
+
+            var fileSystem = SetupStartupScriptFile(filePath, false);
+            var testSubject = CreateTestSubject(filePath, fileSystem: fileSystem.Object);
+            Func<Task> act = async () => await testSubject.Start();
+
+            act.Should().ThrowExactly<FileNotFoundException>().And.Message.Should().Contain(filePath);
+        }
+
+        [TestMethod]
+        public async Task Start_ServerStarts_ReturnsPortNumber()
+        {
+            var scriptPath = CreateScriptThatPrintsPortNumber(123);
+            var startupScriptPath = " "; // pass whitespace so it would not fail NullOrEmpty validation but also do nothing
```

Comment:
No idea how else to simulate this 🙈

Reply:
The `startupScriptPath` is being passed to your dummy node script, isn't it? Does it matter what the arguments are since the script will ignore them anyway?